src/contract_client.rs

```diff
@@ -275,3 +342,182 @@ impl<Type> ContractClient<Type> {
         Ok(tx)
     }
 }
+
+/// A builder to simplify sending smart contract updates.
+pub struct ContractUpdateBuilder {
+    payload:    UpdateContractPayload,
+    sender:     AccountAddress,
+    energy:     Energy,
+    expiry:     Option<TransactionTime>,
+    add_energy: Option<Energy>,
+    nonce:      Option<Nonce>,
+    client:     v2::Client,
+}
+
+impl ContractUpdateBuilder {
+    /// Construct a new builder.
+    pub fn new(
+        client: v2::Client,
+        sender: AccountAddress,
+        payload: UpdateContractPayload,
+        energy: Energy,
+    ) -> Self {
+        Self {
+            payload,
+            sender,
+            energy,
+            expiry: None,
+            add_energy: None,
+            nonce: None,
+            client,
+        }
+    }
+
+    /// Add extra energy to the call.
```

Comment:
Would it make sense to explain in further detail here why/in which case it would be necessary to add extra energy to the transaction?

Reply:
Yes.